index.js

```diff
@@ -1,27 +1,73 @@
 var $ = require('jquery')
+var Mustache = require('mustache')
 
-function AnnotationPoller (loaded) {
+function AnnotationPoller (opts, loaded) {
   var _this = this
+  this.pollInterval = opts.pollInterval || 3000
   this.endpoint = '/api/v1/annotations.js'
   this.annotations = {}
+  this.template = '<li id="annotation-{{id}}" style="{{status}}"><span>{{description}}</span><a href="{{{external-link}}}">{{external-link-text}}</a></li>'
+  this.addonSelector = '#npm-addon-box'
 
   $(document).ready(function () {
-    _this.load(function () {
-      return loaded()
-    })
+    _this.start(loaded)
   })
 }
 
-AnnotationPoller.prototype.load = function (cb) {
+AnnotationPoller.prototype.start = function (loaded) {
+  var _this = this
+  var updating = false
+  this.interval = setInterval(function () {
+    if (updating) return
+    updating = true
+
+    _this.getAnnotations(function () {
+      updating = false
+      _this.renderAnnotations()
+      if (loaded) {
+        loaded()
+        loaded = null
+      }
+    })
+  }, this.pollInterval)
+}
+
+AnnotationPoller.prototype.stop = function () {
+  clearInterval(this.interval)
+}
+
+AnnotationPoller.prototype.getAnnotations = function (cb) {
   var _this = this
   $.getJSON(this.endpoint, function (data) {
     $.each(data, function (i, annotation) {
-      _this.annotations[annotation.uri] = annotation
+      _this.annotations[annotation.id] = annotation
```

Comment:
I was picturing that this would be the `client_id` associated with the OAuth application, which is guaranteed to be unique. This seemed like a good way to ensure that each integration only places one widget.